src/jit/compiler.h

```diff
@@ -1411,8 +1411,14 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     var_types                       GetHfaType(GenTreePtr tree);
     unsigned                        GetHfaSlots(GenTreePtr tree);
 
+    // Only Visual C++ supports inline functions with definitions elsewhere
+#if defined(_MSC_VER_)
     inline var_types                GetHfaType(CORINFO_CLASS_HANDLE hClass);
     inline unsigned                 GetHfaSlots(CORINFO_CLASS_HANDLE hClass);
+#else
+    var_types                GetHfaType(CORINFO_CLASS_HANDLE hClass);
+    unsigned                 GetHfaSlots(CORINFO_CLASS_HANDLE hClass);
+#endif
```

Comment:
We may want to fix the declaration location of these depending on how perf sensitive they are.